File: 003-ArrayDiff/arrayDiff.js
function arrayDiff(a, b) {
    const filteredArray = a.filter(element => !b.includes(element));

    // using set to remove the duplicates.
    return [ ...new Set(filteredArray)]
  }
  
console.log(arrayDiff([1, 2, 2, 3, 3, 4, 4], [2, 3]));

module.exports = arrayDiff;

File: 003-ArrayDiff/arrayDiff.py
def array_diff(a, b):
    
    filtered_array = [element for element in a if element not in b]
    
    # using dictionary to preserve order and remove duplicates.
    return list(dict.fromkeys(filtered_array))

# test cases
print(array_diff([1, 2, 2, 3], [1]))